src/Microsoft.Identity.Web.Certificate/ISigningCredentialsLoader.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.Identity.Abstractions;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Interface for loading signing credentials.
+    /// </summary>
+    public interface ISigningCredentialsLoader
+    {
+        /// <summary>
+        /// Loads SigningCredentials from the credential description.
+        /// </summary>
+        /// <param name="credentialDescription">Credential description.</param>
+        /// <param name="parameters">Optional parameters for loading credentials.</param>
+        /// <returns>SigningCredentials if successful, null otherwise.</returns>
+        Task<SigningCredentials?> LoadSigningCredentialsAsync(
```

Comment:
Not sure this is needed if `CredentialDescription` can alraedy return an X509Certificate2 and potentially a SymmetricKey . It becomes just a convenience method.